roles/galera/tasks/cluster_nodes.yml

```diff
@@ -95,14 +96,21 @@
     owner: mysql
   with_items:
      - "{{ galera_tls }}"
+  no_log: true
   when:
     - galera_ssl | bool
 
+- name: "Check if this is a new installation or not"
+  stat:
+    path: "/etc/pki/mysql/{{ galera_sst_crt_name }}"
+  register: mariadb_new_installation
+
 - name: Create state transfer key
-  shell: cat /etc/pki/mysql/{{galera_server_key_name }} /etc/pki/mysql/{{galera_server_crt_name }} > /etc/pki/mysql/{{ galera_sst_crt_name }}
+  shell: awk 'FNR==1{print ""}1' /etc/pki/mysql/{{ galera_server_key_name }} /etc/pki/mysql/{{galera_server_crt_name }} > /etc/pki/mysql/{{ galera_sst_crt_name }}
+  args:
+    creates: "/etc/pki/mysql/{{ galera_sst_crt_name }}"
   when:
     - galera_ssl | bool
-  changed_when: false
```

Comment:
Is the awk command idempotent?

Reply:
The `awk` command itself not, but task is never executed because the task is checking to see if file `/etc/pki/mysql/{{ galera_sst_crt_name }}` exist. If that is the case, the task is skipped.

Reply:
:+1: